src/foam/nanos/auth/PermissionedPropertyDAO.js

```diff
@@ -48,6 +48,9 @@ foam.CLASS({
       name: 'select_',
       javaCode: `
       if ( x.get("auth") != null ) {
+        if ( predicate != null && ! predicate.authorize(x) ) {
+          return super.select_(x, sink, skip, limit, order,  new foam.mlang.predicate.True());
```

Comment:
Two options:
1. Replace just the offending predicate with TRUE. so AND(EQ, EQ....) -> AND(TRUE, EQ, ... ) 
2. Throw an authentication/authorization exception. 
I don’t think collapsing the entire argument predicate to TRUE is appropriate.

Reply:
I think it makes sense to go for your option 2, because the current idea still allows un-authorized users to figure out whether a permissioned property exists in a model.